src/pages/spawn/spawnHost/SpawnHostActionButton.tsx

```diff
@@ -1,23 +1,110 @@
-import React from "react";
+import React, { useEffect } from "react";
+import { useMutation, useLazyQuery } from "@apollo/client";
 import styled from "@emotion/styled";
 import Button, { Size } from "@leafygreen-ui/button";
 import Icon from "components/icons/Icon";
+import { useBannerDispatchContext } from "context/banners";
+import {
+  UpdateSpawnHostStatusMutation,
+  UpdateSpawnHostStatusMutationVariables,
+  SpawnHostStatusActions,
+  MyHostsQuery,
+  MyHostsQueryVariables,
+} from "gql/generated/types";
+import { UPDATE_SPAWN_HOST_STATUS } from "gql/mutations";
+import { GET_MY_HOSTS } from "gql/queries";
+import { useNetworkStatus } from "hooks";
 import { HostStatus } from "types/host";
 import { MyHost } from "types/spawn";
 
 export const SpawnHostActionButton: React.FC<{ host: MyHost }> = ({ host }) => {
+  const dispatchBanner = useBannerDispatchContext();
+
+  const glyph = mapStatusToGlyph[host.status];
   const action = mapStatusToAction[host.status];
 
-  return action ? (
-    <PaddedButton glyph={<Icon glyph={action} />} size={Size.XSmall} />
-  ) : null;
+  // When the UPDATE_SPAWN_HOST_STATUS mutation occurs the host state is not immediately updated, It gets updated a few seconds later.
+  // Since the GET_MY_HOSTS query on this components parent polls at a slower rate, this component triggers a poll at a faster interval for that
+  // query when it returns an updated host status the polling is halted. This allows the query to poll slowly and not utilize unnecessary bandwith
+  // except when an action is performed and we need to fetch updated data.
+  const [getMyHosts, { startPolling, stopPolling }] = useLazyQuery<
+    MyHostsQuery,
+    MyHostsQueryVariables
+  >(GET_MY_HOSTS, {
+    pollInterval: 1000,
```

Comment:
do we also need to add `fetchPolicy: "network-only"`. I wasn't able to find anything in the apollo docs about what `fetchPolicy` defaults when a pollInterval is set.

Reply:
It most likely goes to network only since there would be no point to poll from  from the cache.